### ISSUE TITLE
fortios_system_ha resource overrides ha settings on synced machines

### ISSUE THREAD

**Issue by agyss** [fortinetdev/terraform-provider-fortios#305]
I'm using the provider with the ha configuration as described by fortigate here: https://github.com/fortinet/azure-templates/blob/main/FortiGate/Active-Passive-SDN/doc/config-provisioning.md

my definitions look like this
```
resource "fortios_system_autoscript" "failover_ha_config" {
  provider    = fortios.failover
  interval    = 1
  name        = "hasetupfailover"
  output_size = 10
  repeat      = 1
  timeout     = 120
  script      = <<EOF
config sys ha
set group-id 1
set group-name Azure-HA
set priority 1
set mode a-p
set hbdev port4 100
set ha-mgmt-status enable
config ha-mgmt-interfaces
    edit 1
      set interface "port1"
      set gateway 10.0.0.1
    next
  end
  set override disable
  set priority 1
  set unicast-gateway 10.0.2.1
  set unicast-hb enable
  set unicast-hb-peerip 10.0.2.10
end

EOF
  start       = "auto"
}

```

```
resource "fortios_system_autoscript" "primary_ha_config" {
  interval    = 1
  name        = "hasetupprimary"
  output_size = 10
  repeat      = 1
  timeout     = 120
  script      = <<EOF
config sys ha
  set group-id 1
  set group-name Azure-HA
  set priority 255
  set mode a-p
  set hbdev port4 100
  set ha-mgmt-status enable
  config ha-mgmt-interfaces
    edit 1
      set interface "port1"
      set gateway 10.0.0.1
    next
    end
    set override disable
    set priority 255
    set unicast-gateway 10.0.2.1
    set unicast-hb enable
    set unicast-hb-peerip 10.0.2.11
  end
EOF
  start       = "auto"
}
```

As I can see in the logs, the initial setup works as expected:
Primary-Forti is configured as desired and uses 10.0.2.11 as unicast-hb-peerip
Secondary-Forti os configured as desired and uses 10.0.2.10 as unicast-hb-peerip

From the logs I know that they find each other and the start exchanging configuration. However, as soon as they finish configuration exchange, the autoscripts are exchanged as well. Unfortunately, the get executed again on each instance, leading to the vpn configuration to be overwritten and effectively the priority as well as the unicast-hb-peerip to become broken.

I'm not sure how to fix this - maybe deletion of the script at the end of execution is possible, but I will have to investigate.

**Comment by agyss** [comment]
adding those lines at the end of the script
```
config system auto-script
delete "hasetupprimary"
```
leads to an error in the provider.

```
Error: Provider produced inconsistent result after apply

When applying changes to fortios_system_autoscript.primary_ha_config,
provider "provider[\"registry.terraform.io/fortinetdev/fortios\"]" produced
an unexpected new value: Root object was present, but now absent.

This is a bug in the provider, which should be reported in the provider's
own issue tracker.
```

So I would argue that ha configuration via terraform is not in a reliable state for production environments.

**Comment by MaxxLiu22** [comment]
Hi @agyss ,

Thank you for raising this issue and sorry for the late reply, after investigating, we plan to go back use fortios_system_ha resource instead of fortios_system_autoscript to fix these issues, and the problems you mentioned(failover FGT runs autoscript after sync and deleting script causes terraform error) are expected, we have a temporary solution to use generic api resource for now. Setting system ha would take a couple of seconds for FOS to react, so waiting a while is expected when you apply ha setting to FGT. Feel free to let me know if you have any other questions or concerns.
```
resource "fortios_json_generic_api" "system_ha" {
  path   = "/api/v2/cmdb/system/ha"
  method = "PUT"
  json   = <<EOF
{
    "group-id": 1,
    "group-name": "Azure-HA",
    "priority": 1,
    "mode": "a-p",
    "hbdev": "port6 100",
    "ha-mgmt-status": "enable",
    "ha-mgmt-interfaces": {
        "id": 1,
        "interface": "port2",
        "gateway": "10.0.0.1"
    },
    "unicast-gateway": "10.0.2.1",
    "unicast-hb": "enable",
    "unicast-hb-peerip": "10.0.2.10"
}
EOF
}
```

Thanks,
Maxx

**Comment by MaxxLiu22** [comment]
Hi @agyss 

This issue has been fixed in our latest Terraform FOS 1.19.0, you may use [fortios_system_ha](https://registry.terraform.io/providers/fortinetdev/fortios/latest/docs/resources/fortios_system_ha) resource to config ha setting now, thank you again for raising this issue. I will go ahead to close this case, you are free to reopen it or another case if you have any other questions.

Thanks,
Maxx
